include/ConfigReader.hpp

```diff
@@ -0,0 +1,49 @@
+#ifndef CONFIG_READER_HPP
+#define CONFIG_READER_HPP
+
+#include <string>
+#include <regex>
+#include <unordered_set>
+#include <filesystem>
+
+/*
+    ConfigReader Class
+    ------------------
+    * Loads simulation config (mesh paths, polynomial degrees, simulation period, time step, Reynolds number) using regex and filesystem.
+    * Validates all required parameters and prompts user if any are missing.
+    * Provides public getters for easy access to these configuration values.
+    ------------------
+*/
+
+
+class ConfigReader
```

Comment:
There already exists a class within deal.II, named `ParameterHandler`, that allows to do this kind of thing. I suggest you look into it.